keel-core/src/main/java/com/netflix/rocket/semver/shaded/DebianVersionComparator.java

```diff
@@ -93,9 +93,17 @@ private int compareMainPart(String s0, String s1) {
       if (i >= mainParts1.length) {
         return 1;
       }
-      int diff = Integer.parseInt(mainParts0[i]) - Integer.parseInt(mainParts1[i]);
-      if (diff != 0) {
-        return diff;
+      boolean isNumeric0 = isNumeric(mainParts0[i]);
```

Comment:
This fixes the `NumberFormatException` we were seeing in the logs.

Reply:
Just to confirm my understanding, the behavior here is:

* if exactly one argument is numeric, the numeric argument is greater
* If neither is numeric: s1 is greater

If neither is numeric, should we do an equality check?

Reply:
TBH, I didn't spend time reviewing this other than it would avoid `NumberFormatException`, because the code is literally copied/pasted as is from Rocket.